app.js

```diff
@@ -23,9 +41,47 @@ app.use(cookieParser());
 app.use(bodyParser.json());
 app.use(express.static(path.join(__dirname, "public")));
 
+app.use(isAuth);
 app.use("/", indexRouter);
 app.use("/users", usersRouter);
 
+io.sockets.on("connection", function (socket, client) {
+  console.log("client connected!")
+  socket.on('room', function(room){
+    socket.join(room);
+    console.log(`Someone joined the room: ${room}`)
+  })
+  socket.on("newMessage", async (data) => {
+    try {
+        console.log(data);
+        Topic.findById(data.topicId, async function(err,Topic){
```

Comment:
Instead of nesting the call, you can simply store the result of first call i.e to find Topic and then use that.

Reply:
Resolved